Common/Data/Market/QuoteBar.cs

```diff
@@ -0,0 +1,420 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Globalization;
+using System.IO;
+using QuantConnect.Logging;
+
+namespace QuantConnect.Data.Market
+{
+    /// <summary>
+    /// QuoteBar class for second and minute resolution data: 
+    /// An OHLC implementation of the QuantConnect BaseData class with parameters for candles.
+    /// </summary>
+    public class QuoteBar : BaseData, IBar
+    {
+        // scale factor used in QC equity/forex data files
+        private const decimal _scaleFactor = 10000m;
+
+        /// <summary>
+        /// Average bid size
+        /// </summary>
+        public long AvgBidSize { get; set; }
+        
+        /// <summary>
+        /// Average ask size
+        /// </summary>
+        public long AvgAskSize { get; set; }
+
+        /// <summary>
+        /// Bid OHLC
+        /// </summary>
+        public Bar Bid { get; set; }
+
+        /// <summary>
+        /// Ask OHLC
+        /// </summary>
+        public Bar Ask { get; set; }
+
+        /// <summary>
+        /// Opening price of the bar: Defined as the price at the start of the time period.
+        /// </summary>
+        public decimal Open { get; set; }
```

Comment:
Agree. I was thinking about that when I was writing the constructors.
That way, in the constructors I would need to define Bid and Ask first, the calculate the Mid, right?